src/v2/Components/FlashBanner/__tests__/FlashBanner.jest.tsx

```diff
@@ -1,32 +1,181 @@
+      data: {
+        me: { id: "woot", canRequestEmailConfirmation: false },
+      },
+      queryString: "?flash_message=expired_token",
+    })
 
     expect(wrapper.text()).toContain("Link expired. Resend verification email.")
   })
+
+  it("returns nothing for an unsupported content code", async () => {
+    const wrapper = await getWrapper({
+      data: {
+        me: { id: "woot", canRequestEmailConfirmation: false },
+      },
+      queryString: "?flash_message=porntipsguzzardo",
+    })
+
+    expect(wrapper.html()).toEqual("")
+  })
+})
+
+describe("Email Confirmation CTA", () => {
+  describe("user cannot request email confirmation", () => {
+    it("user is not prompted to request email confirmation if no logged in user is present", async () => {
+      const wrapper = await getWrapper({ data: { me: null } })
+
+      expect(wrapper.find(Banner).exists()).toBeFalsy()
+    })
+
+    it("user is not prompted to request email confirmation if !me.canRequestEmailConfirmation", async () => {
+      const wrapper = await getWrapper({
+        data: {
+          me: { id: "woot", canRequestEmailConfirmation: false },
+        },
+      })
+
+      expect(wrapper.find(Banner).exists()).toBeFalsy()
+    })
+  })
+
+  describe("user can request email confirmation (me?.canRequestEmailConfirmation=true)", () => {
+    it("user is prompted to request email confirmation in absence of a ?flash_message", async () => {
+      const wrapper = await getWrapper({
+        data: { me: { id: "woot", canRequestEmailConfirmation: true } },
+      })
+
+      expect(wrapper.text()).toContain("Please verify your email address")
+    })
+
+    it("user seeing banner can click to trigger email confirmation message", async () => {
+      const wrapper = await getWrapper({
+        data: { me: { id: "woot", canRequestEmailConfirmation: true } },
+        mutationResults: {
+          sendConfirmationEmail: {
+            confirmationOrError: {
+              unconfirmedEmail: "ceo@blackwater.biz",
+            },
+          },
+        },
+      })
+
+      expect(wrapper.text()).toContain("Please verify your email address")
+
+      wrapper
+        .find("button")
+        .first()
+        .simulate("click")
+      await flushPromiseQueue()
+      wrapper.update()
+
+      expect(wrapper.text()).toContain(
+        "An email has been sent to ceo@blackwater.biz"
+      )
+    })
+
+    describe("with `flash_message` in query string", () => {
+      it("a flash_message indicating an error supercedes the confirmation prompt", async () => {
+        const wrapper = await getWrapper({
+          data: {
+            me: { id: "woot", canRequestEmailConfirmation: true },
+          },
+          queryString: "?flash_message=invalid_token",
+        })
+
+        expect(wrapper.text()).toContain("An error has occurred.")
+      })
+
+      it("user sees an actionable error message to re-trigger verification if flash_message=expired_token", async () => {
```

Comment:
recommend removing this tests for now as it doesn't assert upon anything any more actionable here.